### PR TITLE
feat(pack): add cpp base pack

### PR DESCRIPTION
**Pull request by doupongzeng**
Hello, 

I noticed that there wasn't a package for cpp, so I created one for myself. Perhaps it will be useful to someone else.

### CODE REVIEW

**Comment by Uzaaft** [comment]
Looks good to me. 
@doupongzeng  Should we perhaps add stuff like: `p00f/clangd_extensions.nvim` and  `Civitasv/cmake-tools.nvim`?

File: lua/astrocommunity/pack/cpp/cpp.lua [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    end,
+  },
+  {
+    "williamboman/mason-lspconfig.nvim",
+    opts = function(_, opts) opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, "clangd") end,
+  },
+  {
+    "jay-babu/mason-null-ls.nvim",
+    opts = function(_, opts) opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, "clang-format") end,
+  },
```
Comment (Uzaaft) [inline lua/astrocommunity/pack/cpp/cpp.lua:18]:
```suggestion
  },
  {"p00f/clangd_extensions.nvim",
      ft = { "c", "cpp", "objc", "objcpp", "h", "hpp" },
  }, { "Civitasv/cmake-tools.nvim",
      config = function()
        require("user.cle").cmake_config()
      end,
      ft = { "c", "cpp", "objc", "objcpp", "h", "hpp" },}
```

Comment (Uzaaft) [reply lua/astrocommunity/pack/cpp/cpp.lua:18]: I think the default is that the server is not setup. So in theory we wouldn't need that, no?

Comment (mehalter) [reply lua/astrocommunity/pack/cpp/cpp.lua:18]: yeah and

```lua
init = function() table.insert(astronvim.lsp.skip_setup, "tsserver") end
```

Comment (Uzaaft) [reply lua/astrocommunity/pack/cpp/cpp.lua:18]: Ah got it. I just stole from my old config 😥

Comment (owittek) [reply lua/astrocommunity/pack/cpp/cpp.lua:18]: > yeah and
> 
> 
> 
> ```lua
> 
> init = function() table.insert(astronvim.lsp.skip_setup, "clangd") end
> 
> ```

why `table.insert` over `list_insert_unique`?

Comment (mehalter) [reply lua/astrocommunity/pack/cpp/cpp.lua]: This is a good point. I changed it

**Review by mehalter — COMMENTED** [review]
I went ahead and added `clangd_extensions`, `cmake-tools.nvim` and also standardized the file name

**Review by Uzaaft — APPROVED** [review]
Works like a charm☘️

**Pull request merged**